src/amazon/dsstne/engine/NNNetwork.cpp

```diff
@@ -1900,111 +1905,153 @@ bool NNNetwork::SaveNetCDF(const string& fname)
+uint64_t NNNetwork::GetBufferSize(const string& layer) const
 {
-    NNLayer* pLayer         = _mLayer[layer];
-    if (pLayer == NULL)
+    const auto itr = _mLayer.find(layer);
+    if (itr == _mLayer.end())
     {
         if (getGpu()._id == 0)
+        {
             printf("NNNetwork::GetDeltaBuffer: Unknown layer %s.\n", layer.c_str());
+        }
+
         return 0;
     }
 
-    return pLayer->GetBufferSize();
+    return itr->second->GetBufferSize();
 }
 
-NNWeight* NNNetwork::GetWeight(const string& inputLayer, const string& outputLayer)
+const NNWeight* NNNetwork::GetWeight(const string& inputLayer, const string& outputLayer) const
 {
-    NNLayer* pInputLayer    = _mLayer[inputLayer];
-    NNLayer* pOutputLayer   = _mLayer[outputLayer];
-
-    // Validate parameters
-    if (pInputLayer == NULL)
+    const auto inputLayerItr = _mLayer.find(inputLayer);
+    if (inputLayerItr == _mLayer.end())
     {
         if (getGpu()._id == 0)
+        {
             printf("NNNetwork::GetWeight: Unknown input layer %s.\n", inputLayer.c_str());
+        }
+
         return NULL;
     }
-    if (pOutputLayer == NULL)
+    
+    const auto outputLayerItr = _mLayer.find(inputLayer);
+    if (outputLayerItr == _mLayer.end())
     {
         if (getGpu()._id == 0)
-            printf("NNNetwork::GetWeight: Unknown layer %s.\n", outputLayer.c_str());
+        {
+            printf("NNNetwork::GetWeight: Unknown output layer %s.\n", outputLayer.c_str());
+        }
+
         return NULL;
     }
 
+    const NNLayer *pInputLayer = inputLayerItr->second;
+    const NNLayer *pOutputLayer = outputLayerItr->second;
+
     // Search for matching set of weights
     for (auto p: _vWeight)
+    {
         if ((&(p->_inputLayer) == pInputLayer) && (&(p->_outputLayer) == pOutputLayer))
+        {
             return p;
+        }
+    }
 
     // Report failure to find weights connecting layers
     if (getGpu()._id == 0)
+    {
         printf("NNNetwork::GetWeight: No set of weights connecting layer %s to layer %s.\n", outputLayer.c_str(), outputLayer.c_str());
+    }
+
     return NULL;
 }
 
-NNFloat* NNNetwork::GetWeightBuffer(const string& inputLayer, const string& outputLayer)
+const NNFloat* NNNetwork::GetWeightBuffer(const string& inputLayer, const string& outputLayer) const
 {
-    NNLayer* pInputLayer    = _mLayer[inputLayer];
-    NNLayer* pOutputLayer   = _mLayer[outputLayer];
-
-    // Validate parameters
-    if (pInputLayer == NULL)
+    const auto inputLayerItr = _mLayer.find(inputLayer);
+    if (inputLayerItr == _mLayer.end())
     {
         if (getGpu()._id == 0)
-            printf("NNNetwork::GetWeightBuffer: Unknown input layer %s.\n", inputLayer.c_str());
+        {
+            printf("NNNetwork::GetWeight: Unknown input layer %s.\n", inputLayer.c_str());
+        }
+
         return NULL;
     }
-    if (pOutputLayer == NULL)
+    
+    const auto outputLayerItr = _mLayer.find(inputLayer);
```

Comment:
it is a copy/paste bug: it should be
const auto outputLayerItr = _mLayer.find(outputLayer);